pkg/controllers/node/node_controller.go

```diff
@@ -182,6 +197,119 @@ func (c *NodeController) syncDelete() error {
 	return c.syncDeleteEtcd()
 }
 
+// isAutoHostendpoint determines if the given hostendpoint is managed by
+// kube-controllers.
+func isAutoHostendpoint(h *api.HostEndpoint) bool {
+	v, ok := h.Labels[hepCreatedLabelKey]
+	return ok && v == hepCreatedLabelValue
+}
+
+// createHostendpoint creates an auto hostendpoint for the specified node.
+func (c *NodeController) createHostendpoint(n *api.Node) (*api.HostEndpoint, error) {
+	// Create a set of labels from the node labels and include the
+	// special label marking the hep as created by us.
+	hepLabels := make(map[string]string, len(n.Labels))
+	hepLabels[hepCreatedLabelKey] = hepCreatedLabelValue
+	for k, v := range n.Labels {
+		hepLabels[k] = v
+	}
+
+	hep := &api.HostEndpoint{
+		TypeMeta: metav1.TypeMeta{Kind: "HostEndpoint", APIVersion: "v3"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   n.Name,
```

Comment:
I'm not 100% sure about just using the node name - what happens if existing users have already created HEPs with this name, (which seems plausible)?

We might want to make it more unique in some way in order to protect that. Maybe `fmt.Sprintf("%s-auto-hep", n.Name)`?

Reply:
Yeah good point.